scripts/merge_truth_per_tract.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+
+"""
+Merge truth catalogs for a given tract and match to object catalog
+Author: Yao-Yuan Mao
+"""
+import os
+import re
+from argparse import ArgumentParser, RawTextHelpFormatter
+
+import numpy as np
+from astropy.coordinates import SkyCoord
+import pandas as pd
+
+__all__ = ["merge_truth_per_tract", "match_object_with_merged_truth"]
+
+
+def merge_truth_per_tract(input_dir, validate=False, silent=False, **kwargs):
+    """Merge all the truth catalogs in one single tract directory
+
+    Parameters
+    ----------
+    input_dir : str
+        Path to the input tract directory
+
+    Optional Parameters
+    ----------------
+    validate : bool, optional (default: False)
+        If true, check the tract column has only one value
+    silent : bool, optional (default: False)
+        If true, turn off most printout.
+    """
+    my_print = (lambda x: None) if silent else print
+
+    my_print("Merging all files in", input_dir)
+    files_to_merge = sorted(os.listdir(input_dir))
+    df_to_merge = []
+    for filename in files_to_merge:
+        my_print("Loading", filename)
+        df = pd.read_parquet(os.path.join(input_dir, filename))
+        m = re.search(r"_hp(\d+)\b", filename)
+        if m is not None:
+            hp = int(m.groups()[0])
+            df["cosmodc2_hp"] = hp
+            df["cosmodc2_id"] = df["id"]
+            df["id"] = df["id"].astype(str)
+        else:
+            df["cosmodc2_hp"] = -1
+            df["cosmodc2_id"] = -1
+        df["is_sn"] = True if filename.startswith("sn_") else False
```

Comment:
Right. I am not sure how general we can make the code because it is designed to tackle this specific usage.  I did change this function a bit with a new `truth_types` argument, and explains in the docstring that we expect the input file names to have prefix as specified in `truth_types`. In the data product, I also added a new column `truth_type`, with 1=galaxy; 2=star; 3=sn.